attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -179,11 +181,19 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
         Ok(())
     }
 
-    async fn generate_evidence(&self, runtime_data: String) -> Result<String> {
+    async fn generate_evidence(
+        &self,
+        runtime_data: String,
+        challenge_extra_params: String,
+    ) -> Result<String> {
         let mut hasher = Sha384::new();
         hasher.update(runtime_data);
 
-        let ehd = hasher.finalize().to_vec();
+        let mut ehd = hasher.finalize().to_vec();
+        // IBM SE uses challenge_extra_params as runtime_data to pass attestation_request
+        if challenge_extra_params.chars().count() > 0 {
+            ehd = challenge_extra_params.into_bytes();
+        }
```

Comment:
do we want to have this as default behaviour? other TEE  impls might have `extra_params` set and still want to use the hashed runtime_data.

we have the `tee` available on the calling function site, so we can do a special case handling.

Reply:
Yes, do some special for IBM SE and other TEE based on `tee` is alternative and I think it's hood also.

Reply:
Fixed